src/commands/install.rs

```diff
@@ -75,16 +75,50 @@ pub fn install(
     }
 
     // Update local manifest
+    // -------------------------------------------------------------------------
     let local_manifest_path = config.midenup_home.join("manifest").with_extension("json");
+    {
+        // Check if the installed channel needs to marked as stable
+        let mut channel_to_save = if is_latest_stable {
+            let mut modifiable = channel.clone();
+            modifiable.alias = Some(ChannelAlias::Stable);
+            modifiable
+        } else {
+            channel.clone()
+        };
+
+        // If a component was installed with --branch, then write down the
+        // current commit. This is used on updates to check if any new commits
+        // were pushed since installation.
+        // NOTE: To check the latest commit we're using git cli instead. Would
```

Comment:
I think it is fine to expect `git` to be on the user's PATH, and just shell out to it. It'll be far simpler, and we don't need to effectively ship a git binary inside `midenup`.